src/components/webcams/streamers/WebrtcGo2rtc.vue

```diff
@@ -0,0 +1,205 @@
+        }
+
+        return this.convertUrl(url.toString(), this.printerUrl)
+    }
+
+    // stop and restart the video if the url changes
+    @Watch('url')
+    changedUrl() {
+        this.terminate()
+        this.start()
+    }
+
+    get expanded(): boolean {
+        return this.$store.getters['gui/getPanelExpand']('webcam-panel', this.viewport) ?? false
+    }
+
+    // start or stop the video when the expand state changes
+    @Watch('expanded', { immediate: true })
+    expandChanged(newExpanded: boolean): void {
+        if (!newExpanded) {
+            this.terminate()
+            return
+        }
+
+        this.start()
+    }
+
+    log(msg: string, obj?: any) {
+        if (obj) {
+            window.console.log(`[WebRTC go2rtc] ${msg}`, obj)
+            return
+        }
+
+        window.console.log(`[WebRTC go2rtc] ${msg}`)
+    }
+
+    // webrtc player methods
+    // adapted from https://github.com/AlexxIT/go2rtc/blob/master/www/webrtc.html
+
+    start() {
+        this.log('connecting to ' + this.url)
+        this.status = 'connecting'
+
+        this.pc = new RTCPeerConnection({
+            iceServers: [{ urls: 'stun:stun.l.google.com:19302' }],
+        })
+
+        let localTracks: MediaStreamTrack[] = []
+        const kinds = ['video', 'audio']
+        kinds.forEach((kind: string) => {
+            const track = this.pc?.addTransceiver(kind, { direction: 'recvonly' }).receiver.track
+            if (track) localTracks.push(track)
+        })
+        this.video.srcObject = new MediaStream(localTracks)
+
+        this.ws = new WebSocket(this.url)
+        this.ws.addEventListener('open', () => this.onWebSocketOpen())
+        this.ws.addEventListener('message', (ev) => this.onWebSocketMessage(ev))
+        this.ws.addEventListener('close', () => this.onWebSocketClose())
+    }
+
+    onWebSocketOpen() {
+        this.log('open')
+        this.status = 'connected'
+
+        this.pc?.addEventListener('icecandidate', (ev) => {
+            if (!ev.candidate) return
+            const msg = { type: 'webrtc/candidate', value: ev.candidate.candidate }
+            this.ws?.send(JSON.stringify(msg))
+        })
+
+        this.pc
+            ?.createOffer()
+            .then((offer) => this.pc?.setLocalDescription(offer))
+            .then(() => {
+                const msg = { type: 'webrtc/offer', value: this.pc?.localDescription?.sdp }
+                this.ws?.send(JSON.stringify(msg))
+            })
+    }
+
+    onWebSocketMessage(ev: MessageEvent) {
+        const msg = JSON.parse(ev.data)
+
+        if (msg.type === 'webrtc/candidate') {
+            this.pc?.addIceCandidate({ candidate: msg.value, sdpMid: '0' })
+        } else if (msg.type === 'webrtc/answer') {
+            this.pc?.setRemoteDescription({ type: 'answer', sdp: msg.value })
+        }
+    }
+
+    onWebSocketClose() {
+        this.log('close')
+        this.status = 'disconnected'
+        this.scheduleRestart()
```

Comment:
If I comment out this line then the disconnection issues go away, I think the websocket close get called for other reasons that a true disconnection, I think this code needs to be adjusted a bit to account for that.

Reply:
the complete function (onWebSocketClose) will only be executed, when the WebSocket was already closed. so this cannot be the real issue, but i think i forget to cancel the scheduleRestart, after the first open event of the WebSocket.

Reply:
got it.  I can test out the changes once you update the code.

Reply:
I am not able to build this project yet, (no time to update by build env), but if I put a break point on this code the webrtc continues to stream fine. I will attach a video for this example. I am still trying to get to the bottom of this..